packages/flutter/lib/src/material/date.dart

```diff
@@ -203,15 +476,16 @@ typedef SelectableDayPredicate = bool Function(DateTime day);
 ///  * [showDateRangePicker], which displays a dialog that allows the user to
 ///    select a date range.
 @immutable
-class DateTimeRange {
-  /// Creates a date range for the given start and end [DateTime].
+@optionalTypeArgs
+class DateTimeRange<T extends DateTime> {
```

Comment:
What's the reason for this? I have a guess.

Reply:
This is to better support subtypes of DateTime, such as [NepaliDateTime](https://pub.dev/documentation/nepali_utils/latest/nepali_utils/NepaliDateTime-class.html).